app/src/components/formatted-number-input.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { Control, Controller } from "react-hook-form";
+import {
+  InputGroup,
+  InputRightAddon,
+  NumberInput,
+  NumberInputField,
+  NumberInputProps,
+} from "@chakra-ui/react";
+
+interface FormattedNumberInputProps extends NumberInputProps {
+  control: Control<any, any>;
+  name: string;
+  defaultValue?: number;
+  isDisabled?: boolean;
+  placeholder?: string;
+  children?: React.ReactNode;
+  miniAddon?: boolean;
+  testId?: string;
+}
+
+function FormattedNumberInput({
+  control,
+  testId,
+  name,
+  defaultValue = 0,
+  isDisabled = false,
+  children, // Accepts children for InputRightAddon
+  placeholder,
+  miniAddon,
+  ...rest
+}: FormattedNumberInputProps) {
+  const format = (nval: number | string) => {
+    let val = parseInt(nval as string);
+    if (val === undefined || val === null || isNaN(val)) return "";
+    return val.toLocaleString();
+  };
+
+  const parse = (val: string) => val.replace(/,/g, "");
```

Comment:
Problem: this depends on the selected locale. Other languages have different decimal and thousands separators (e.g. German has , as decimal and . for thousands, so reversed from English).

Can we let an i18n/ locale library do this work for us instead of having to implement this ourselves?

Otherwise we will need to have the separators for each language in the i18n util.ts file, but this is clearly logic that others have implemented before and lut into libraries (e.g. i18next).

Reply:
We can make this a new ticket if necessary, but this needs to be dealt with rather urgently as other language users will be confused here.

Reply:
I'll localize